include/picongpu/simulation_defines/param/isaac.param

```diff
@@ -17,27 +17,44 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
+/** @file
+ *
+ * Definition which native fields and density fields of particles will be
+ * visualizable with ISAAC. ISAAC is an in-situ visualization library with which
+ * the PIC simulation can be observed while it is running avoiding the time
+ * consuming writing and reading of simulation data for the classical post
+ * processing of data.
+ * 
+ * ISAAC can directly visualize natives fields like the E or B field, but
+ * density fields of particles need to be calculated from PIConGPU beforehand
+ * which slightly increases the runtime.
```

Comment:
the runtime -> the memory consumption

Reply:
Really? The calculation is for free? And I thought the memory is hold back for such fields anyway?

Reply:
Of course the calcs are also not for free, but the more severe point a user has to be aware of is that the memory is less, because it influences load balancing/GPU partitioning, global simulation size, etc.

But when I think about it: you use the regular FieldTmp slots for your derived fields and only one at a time, so that does not change the memory. so forget what I said.

Reply:
Well... Technically as I need to access all fields **at the same time** I copy them nevertheless. So yes, more particles need more memory. But this memory is not stolen from PIConGPU, as PIConGPU reserves afair 400 MB for such cases. I don't know where FieldTmp resists, but ISAAC's copies are part of this 400 MB and if too many particles are activated ISAAC will crash the simulation while allocating memory for the copies... I should mention this, also where to increase the amount of not-libPMacc-handled-memory. *cough*

Reply:
I remember something like all the GPU memory is allocated __after__ all plugins are initialized. Is this true? In that case ISAAC will indeed steal memory from PIConGPU - but not crash.

Reply:
Yes plugins allocated its memory first and the rest is for the particles. Therefore each *non persistent* field reduce the available memory for particles.

Reply:
Okay, will document it this way